Public login

DennisWei9898 opened the pull request:
新增即時顯示登入的使用者列表，登入和登出時會自動推送系統訊息到聊天視窗中

DennisWei9898 committed:
study login feat in socket.io

DennisWei9898 committed:
Merge remote-tracking branch 'origin' into public_login

# Conflicts:
#	app.js

DennisWei9898 committed:
try to bind socket with connect Users

DennisWei9898 committed:
feat: public room show online & offline users

DennisWei9898 committed:
fix conflict

hazelisles committed:
Merge branch 'master' into public_login

hazelisles commented:
Fix 2 conflicting files
controllers/socketController.js
public/javascripts/check.js

1. 將 socketController 的 console.log user login 刪掉 => 功能做好就不需要了
2. 刪除 .map() 前的 await => map語法是同步執行加await是沒有意義的 
3. 刪除 onlineUsers.length 前的 await => .length語法也是同步執行

hazelisles committed:
disconnect code optimization

刪除 console.log('a user disconnected')
onlineUsers.length 為同步語法 前方 await 並無用處 故刪除
disconnect 的 code 並沒有用到任何非同步語法 故不需要寫成 async/await 因此刪除 async/await

hazelisles committed:
fix unnecessary changes

hazelisles reviewed (approved):
達成顯示上線使用者功能 ✔️ 

onlineUsers.find((item) => { return item.id === idFromSession })
find語法同filter語法，寫法可簡化為
onlineUsers.find(item => item.id === idFromSession)
箭頭函式只有一行的話，是自帶return效果的，故不用再寫return
其他code的修改在上一個comment
